### DISCUSSION TITLE
Touchscreen requires double-clicking to single-click when "Confine Mouse" and/or "High-precision mouse" is enabled

### DISCUSSION THREAD

**Discussion by ItsJustSomeDude** [ppy/osu#22159] [Q&A]
This affects Standard Gameplay and the Main Menu.  I can tap on something (like the osu! cookie) and the cursor will move there, but will not do the click action.  If I tap the cookie again, it will then advance to the next screen.  The same happens with any of the main menu's buttons.

What is strange is that it knows that I clicked, because the cursor will go into it's spinny/drag mode (I don't know what this is called).

In Gameplay, if I want to hit a note, I have to tap once to move the cursor there, then another time to actually hit it.

It took a lot of fiddling before I found that "Confine cursor" and "High-precision mouse" both cause this.

Touch is enabled in settings.

High-precision On, Confine Never: Broken as above.
High-precision On, Confine Always: Broken as above.
High-precision Off, Confine Never: Works perfectly!
High-precision Off, Confine Always: Broken as above.

Gnome + Linux + Wayland.

I didn't see anything relevant, but...
[logs.zip](https://github.com/ppy/osu/files/10405186/logs.zip)
[logs2.zip](https://github.com/ppy/osu/files/10405318/logs2.zip)

**Comment by JacksonChen666** [comment]
> What is strange is that it knows that I clicked, because the cursor will go into it's spinny/drag mode (I don't know what this is called).

"Rotate cursor when dragging" is the setting name, so the mouse was considered to be "dragging" (hold click + move cursor)

> In Gameplay, if I want to hit a note, I have to tap once to move the cursor there, then another time to actually hit it.

that's what i experienced, and after using the settings you mentioned worked, touchscreen gameplay is actually possible (tap once for 1 note instead of twice). summary: reproducible

**Comment by Kaerulans** [comment]
I'm having the same issue. This also happens in full-screen, even if settings are set to High-precision Off, Confine Never

**Comment by JacksonChen666** [comment]
> This also happens in full-screen

check your confine settings again, i couldn't change it because it was in full screen and i had to go borderless

**Reply by Kaerulans** [reply]
Yeah, confine settings can't be changed in fullscreen, but the bug of having to double-click reproduces in fullscreen for me

**Reply by JacksonChen666** [reply]
> confine settings can't be changed in fullscreen, but the bug of having to double-click reproduces in fullscreen for me

if the confine settings can't be changed in fullscreen, you can't change it. have you tried borderless with confine off?

**Reply by Kaerulans** [reply]
Borderless with confine off works fine. What I'm saying is that in fullscreen the bug is still present, regardless of settings.

**Reply by Espionage724** [reply]
I disabled high-precision mouse and didn't adjust confine, but did play fullscreen; confine gets enabled while fullscreen, leading to tap issues?

I was on Linux yesterday and noticed taps were odd but couldn't figure it out (disabling high-precis improved a bit but not fully)
